modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceCheckout.testcase

```diff
@@ -159,6 +159,98 @@ definition {
 			value1 = "Success! Your order has been processed.");
 	}
 
+	@description = "This is a test for COMMERCE-9052. Checkout process from mini-cart with buyer2 for different kind of orders made by buyer1"
+	@priority = "5"
+	test AddOrderFromAUserAndCheckoutFromAnotherUserViaMiniCart {
+		property custom.properties = "jsonws.web.service.paths.excludes=${line.separator}company.security.strangers.verify=false";
+		property portal.acceptance = "false";
+
+		task ("Given a Minium site with a two buyer users") {
+			CommerceAccelerators.initializeNewSiteViaAccelerator(siteName = "Minium");
+
+			CommerceEntry.addAccountEntryUser(
+				accountName = "Commerce Account",
+				accountType = "Business",
+				agreeToTermsAndAnswerReminderQuery = "true",
+				createNewAccount = "true",
+				requireReset = "false",
+				userEmailAddress = "buyer@liferay.com",
+				userFirstName = "Buyer",
+				userLastName = "User",
+				userRole = "Buyer",
+				userScreenName = "buyeruser",
+				userSiteMembership = "Minium");
+
+			CommerceEntry.addAccountEntryUser(
+				accountName = "Commerce Account",
+				agreeToTermsAndAnswerReminderQuery = "true",
+				createNewAccount = "false",
+				requireReset = "false",
+				userEmailAddress = "buyer2@liferay.com",
+				userFirstName = "Buyer",
+				userLastName = "User 2",
+				userRole = "Buyer",
+				userScreenName = "buyeruser2",
+				userSiteMembership = "Minium");
+		}
+
+		task ("Given buyer user 1 logs in") {
+			User.logoutPG();
+
+			CommerceLogin.miniumLogin(
+				password = "test",
+				urlAppend = "web/minium",
+				userEmailAddress = "buyer@liferay.com");
+		}
+
+		task ("And given a product is added to the cart") {
+			FrontStore.addInitializerProductToCart(
+				productName = "U-Joint",
+				productQuantity = "1");
+
+			WaitForElementPresent(locator1 = "CommerceEntry#ACTIVE_ORDER_ID");
+
+			var activeOrderId = selenium.getText("CommerceEntry#ACTIVE_ORDER_ID");
+		}
+
+		task ("When buyer user 2 logs in navigating in Minium site") {
+			User.logoutPG();
+
+			CommerceLogin.miniumLogin(
+				password = "test",
+				urlAppend = "web/minium",
+				userEmailAddress = "buyer2@liferay.com");
+		}
+
+		task ("And opens the mini cart to see the active order of buyer user 1") {
+			AssertTextEquals(
+				locator1 = "CommerceEntry#ACTIVE_ORDER_ID",
+				value1 = "${activeOrderId}");
+		}
+
+		task ("And by clicking Submit button, comletes the checkout process") {
+			CommerceNavigator.gotoMiniCart();
+
+       		Button.click(button = "Submit");
```

Comment:
This will probably not work on CI because of the known resolution issue that affects the Minium sidebar on CI.
Please consider adding a separate macro for these lines.
Add a `ScrollBy(value1 = "0, 1000");` that will scroll the page down and move the sidebar that covers the Submit button. Thanks.